client/src/main/scala/skuber/api/client/KubernetesClient.scala

```diff
@@ -216,10 +229,12 @@ trait KubernetesClient {
     * Watch a specific object resource continuously. This returns a source that will continue to produce
     * events on any updates to the object even if the server times out, by transparently restarting the watch as needed.
     * @param obj  the object resource to watch
+    * @param pool reuse a skuber pool for querying the server if any or create a new one
     * @tparam O the type of the resource e.g Pod
-    * @return  A future containing an Akka streams Source of WatchEvents that will be emitted
+    * @return  A future containing an Akka streams Source of WatchEvents that will be emitted where the materialized
+    *          value is a pair of the skuber pool used and the underlying Akka host connection pool used, if any.
     */
-  def watchContinuously[O <: ObjectResource](obj: O)(implicit fmt: Format[O], rd: ResourceDefinition[O], lc: LoggingContext): Source[WatchEvent[O], _]
+  def watchContinuously[O <: ObjectResource](obj: O, pool: Option[Pool[WatchSource.Start[O]]])(implicit fmt: Format[O], rd: ResourceDefinition[O], lc: LoggingContext): Source[WatchEvent[O], (Pool[WatchSource.Start[O]], Option[Http.HostConnectionPool])]
```

Comment:
The current API completely hides the fact that there's a pool involved and therefore prevents doing anything with it:
- reuse the same across multiple calls (see the PiJobsSequential example)
- create an independent pool for each call (see the PiJobsParallel example)
- shutting down a pool when it is no longer needed

If you're thinking of refactoring this API, it would be good to keep the above use cases in mind; the distinction between sequential & parallel invocations of long-duration operations is very important for us.

I don't know how to support these use cases without changing the API in some incompatible way. Please advise how you want to proceed. I'm OK in deferring action until you refactor the API as you mentioned.